File: simulator.py
import requests
import time

# Update this with your actual URL (check AdmsPushApi/Properties/launchSettings.json)
SERVER_URL = "http://localhost:5122/iclock/cdata" 
DEVICE_SN = "SIMULATOR_001"

def simulate_swipe(user_id):
    timestamp = time.strftime("%Y-%m-%d %H:%M:%S")
    # ZKTeco format: ID \t Timestamp \t Status \t VerifyMode \t ...
    log_data = f"{user_id}\t{timestamp}\t0\t1\t0\t0\n"
    
    print(f"Pushing swipe for User {user_id}...")
    try:
        response = requests.post(f"{SERVER_URL}?SN={DEVICE_SN}", data=log_data)
        if response.text.strip() == "OK":
            print("Successfully pushed to server.")
        else:
            print(f"Server responded with: {response.text}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    print(f"Starting Simulator for SN: {DEVICE_SN}")
    # 1. Perform Handshake
    try:
        requests.get(f"{SERVER_URL}?SN={DEVICE_SN}")
    except Exception as e:
        print(f"Handshake failed: {e}")
    
    # 2. Simulate some swipes
    simulate_swipe("101")
    time.sleep(2)
    simulate_swipe("102")

File: simulator.js
// Node.js Simulator for ADMS Push SDK
// Run with: node simulator.js

const SERVER_URL = "http://localhost:5122/iclock/cdata";
const DEVICE_SN = "SIMULATOR_001";

async function simulateSwipe(userId) {
    const timestamp = new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '');
    // ZKTeco format: ID \t Timestamp \t Status \t VerifyMode \t ...
    const logData = `${userId}\t${timestamp}\t0\t1\t0\t0\n`;

    console.log(`Pushing swipe for User ${userId}...`);
    try {
        const response = await fetch(`${SERVER_URL}?SN=${DEVICE_SN}`, {
            method: 'POST',
            body: logData
        });
        const text = await response.text();
        if (text.trim() === "OK") {
            console.log("Successfully pushed to server.");
        } else {
            console.log(`Server responded with: ${text}`);
        }
    } catch (e) {
        console.error(`Error: ${e.message}`);
    }
}

async function start() {
    console.log(`Starting Node.js Simulator for SN: ${DEVICE_SN}`);

    // 1. Perform Handshake
    try {
        await fetch(`${SERVER_URL}?SN=${DEVICE_SN}`);
    } catch (e) {
        console.error(`Handshake failed: ${e.message}`);
    }

    // 2. Simulate some swipes
    await simulateSwipe("101");
    setTimeout(async () => {
        await simulateSwipe("102");
    }, 2000);
}

start();
